nexus/transport.go

```diff
@@ -78,6 +80,43 @@ type Transport interface {
 	//
 	// NOTE: Experimental
 	Close() error
+
+	// UnimplementedTransport must be embedded into any [Transport] implementation for future compatibility.
+	// It implements all methods on the [Transport] interface, returning unimplemented errors if they are not implemented by
+	// the embedding type.
+	mustEmbedUnimplementedTransport()
+}
+
+// TransportError indicates a client encountered something unexpected in the server's response.
+type TransportError struct {
+	// Error message.
+	Message string
+	// Optional failure that may have been embedded in the response.
+	Failure *Failure
+	// Additional transport specific details.
+	// For HTTP, this would include the HTTP response. The response body will have already been read into memory and
+	// does not need to be closed.
+	Details any
+}
+
+// Error implements the error interface.
+func (e *TransportError) Error() string {
+	return e.Message
+}
+
+func newTransportError(message string, response *http.Response, body []byte) error {
```

Comment:
```suggestion
func newHTTPTransportError(message string, response *http.Response, body []byte) error {
```

since this is for the HTTP transport right? the gRPC transport won't use this I think